src/Components/Components/src/Virtualization/InfiniteScroll.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// Provides functionality for rendering a virtualized, infinite list of items.
+    /// </summary>
+    /// <typeparam name="TItem">The <c>context</c> type for the items being rendered.</typeparam>
+    public sealed class InfiniteScroll<TItem> : VirtualizeBase<TItem>
+    {
+        private readonly ConcurrentQueue<TItem> _loadedItems = new ConcurrentQueue<TItem>();
+
+        private int _fetchingItems;
+
+        protected override int ItemCount => InfiniteScrollFooter == null ? _loadedItems.Count : _loadedItems.Count + 1;
+
+        /// <summary>
+        /// Gets or sets the list item template.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<TItem> Item { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the infinite scroll footer template.
+        /// </summary>
+        [Parameter]
+        public RenderFragment? InfiniteScrollFooter { get; set; }
+
+        /// <summary>
+        /// Gets or sets the function that provides items for the inifinite scroll.
+        /// </summary>
+        [Parameter]
+        public Func<Range, Task<IEnumerable<TItem>>> ItemsProvider { get; set; } = default!;
```

Comment:
Rather than telling the developer how many to fetch, would it work to tell them only the start index and let them fetch as many as they want?

If the per-item height can vary anyway, we don't know the desired number to fetch.